### ISSUE TITLE
fn.name and fn.length?

### ISSUE THREAD

**Issue by jprichardson** [RyanZim/universalify#1]
Piggybacking from this comment... https://github.com/jprichardson/node-fs-extra/pull/403#issuecomment-293963007

Why wouldn't you want to set `fn.name` and `fn.length`? Stated another way, why wouldn't you want to take the implementation from https://github.com/paulmillr/micro-promisify/blob/600daee25abc75af3cef8145ef6eb55aafdd2d54/index.js#L11? I'm not saying that implementation is correct, I'm just interested in your reasoning.

**Comment by RyanZim** [comment]
OK, perhaps I wasn't clear enough. I do want to set `fn.name`. I'll be fixing that as soon as I can.

---

Regarding `fn.length`: Suppose we have this function:
```js
function foo (a, b, callback) { ... }
```
We want to universalify it:
```js
const newFoo = universalify.fromCallback(foo)
```
Now, we can use it like this:
```js
newFoo(1, 2)
.then(..)
.catch(..)
```
or like this:
```js
newFoo(1, 2, (err, res) => {
  // Do something
})
```
Note that in the first example, we called `newFoo` with 2 arguments. In the second example, we called it with 3.

micro-promisify doesn't support calling the transformed functions with callbacks, so it sets `fn.length` to `Math.max(0, fn.length - 1)`. This logic doesn't apply to universalify, since it's still possible to call the function with a callback.

However, if we keep the old `fn.length` value, this might be confusing for promise users. Also, should functions passed to `fromPromise` have their length bumped by 1 or not?

Therefore, I was thinking of just leaving `fn.length` `undefined`.
